cpp/arcticdb/version/symbol_list.cpp

```diff
@@ -166,13 +177,24 @@ bool SymbolList::can_update_symbol_list(const std::shared_ptr<Store>& store,
             const folly::Range<KeyVectorItr>& keys) {
         SYMBOL_LIST_RUNTIME_LOG("Loading symbols from symbol list keys");
         bool read_compaction = false;
+        int uncompacted_keys_found = 0;
         CollectionType symbols{};
         for(const auto& key : keys) {
             if(key.id() == compaction_id) {
                 read_list_from_storage(store, key, symbols);
                 read_compaction = true;
             }
             else {
+                uncompacted_keys_found++;
+                if (uncompacted_keys_found == this->max_delta_ * 2 && !warned_expected_slowdown_) {
```

Comment:
How did you choose `max_delta_ * 2` as the heuristic? Seems rather small, is this really so slow with 1000 symbols? 

Again, what's the point of the `warned_expected_slowdown_` state?

Reply:
At default settings `max_delta_ * 2` is the minimal value that will result in at least two `list_object` calls. Otherwise though it's a heuristic that I've mostly plucked out of thin air...

> is this really so slow with 1000 symbols?

Depends. One user is saying that it's taking minutes with ~1,400 so I think the answer is yes :/ 

> Again, what's the point of the warned_expected_slowdown_ state?

If the user doesn't have write privileges this will repeat for every invocation. I thought that'd be annoying. Happy to remove if you think this checking is unnecessary?

Reply:
I didn't appreciate the point about write privs, so I think what you've done makes sense.

I think we might need to tune the heuristic at some point but I guess we can see how it goes.